app/src/main/java/org/schabi/newpipe/local/playlist/LocalPlaylistManager.java

```diff
@@ -43,28 +42,34 @@ public Maybe<List<Long>> createPlaylist(final String name, final List<StreamEnti
         }
         final StreamEntity defaultStream = streams.get(0);
         final PlaylistEntity newPlaylist =
-                new PlaylistEntity(name, defaultStream.getThumbnailUrl(), false);
+                new PlaylistEntity(name, false, defaultStream.getUid());
 
-        return Maybe.fromCallable(() -> database.runInTransaction(() ->
-                upsertStreams(playlistTable.insert(newPlaylist), streams, 0))
-        ).subscribeOn(Schedulers.io());
+        return Maybe.fromCallable(() -> database.runInTransaction(() -> {
+                    final List<Long> streamIds = streamTable.upsertAll(streams);
+                    newPlaylist.setThumbnailStreamId(streamIds.get(0));
```

Comment:
I don't think you need to call `setThumbnailStreamId` again, haven't you already called `defaultStream.getUid()` above? Upsert shouldn't change ids afaik. (this would mean that changes to `insertJoinEntities` could be reverted)

Reply:
Unfortunately, you have to set it again after upserting the stream.
Streams have a default `uid` value of `0`, so `0` is always returned in the first `defaultStream.getUid()` method. Only after the stream has been upserted it is given a unique ID. Therefore, you must first upsert it and then set the `streamId` of the `PlaylistEntity`.

Reply:
Then can't `newPlaylist` be created here, directly with the correct `streamId`?